### DISCUSSION TITLE
Lidar and Optic flow

### DISCUSSION THREAD

**Discussion by flyinginger** [iNavFlight/inav-configurator#1209] [Ideas]
Enjoining on a build the Matek lidar optic-flow device...
**Feature request : config tab/ dedicated menu...**
Here some CLI command that i'm tweaking to tune the behaviours...flight modes: AltHold + angle and PosHold + angle. Surface on a separate switch to modify, gps with mag active

MATEKs recommandation applied :
set nav_mc_vel_z_p = 150
set nav_mc_vel_z_d = 25
set nav_mc_vel_z_i = 250
set nav_mc_vel_xy_p = 80
set nav_mc_vel_xy_i = 40
set nav_mc_vel_xy_d = 60
set nav_mc_pos_xy_p = 50

CLI to be explored still at default :

Set inav_w_z_surface_p
Set inav_w_z_surface_v
nav_mc_hover_thr
nav_fw_loiter_radius
Set inav_baro_epv = 
inav_w_z_baro_p =  

inav_max_surface_altitude = 
nav_max_terrain_follow_alt =

inav_w_z_surface_p =  
inav_w_z_surface_v = 

opflow_scale =
inav_w_xy_flow_p = 
inav_w_xy_flow_v = 

nav_mc_pos_z_p

nav_user_control_mode
rangefinder_median_filter

inav_reset_altitude (first arm is default, will do every arm next test)

What are your recommendations ?
For now, having issue with "yoyo" behaviour and the fact that i need to land, disarm, arm in order to activate/deactivate surface mode... difference beteween terrain and surface CLI command sound the same to me...

Let's make op flow and lidar awesome !

**Comment by Some-Won-1** [comment]
Yes, I am out of UART ports and the TFMini-S (Lidar) can be switched over to use the I2C bus, but its connection is not yet ready for prime time in 4.0 firmware.  Interesting...?

One thing I surmised with the Configurator is that it DOES NOT show in Configuration "Sensors & Buses" if the connection is from a UART or an I2C interface. It would be very helpful even if the combo was re titled different to indicate such. Connection: Connection.

I also see indication of units inside the input box that are disconnected from users selection compared to the TITLE.

I vote make it all metric and be done.

**Comment by krasaee-spire** [comment]
Folks, any thoughts on this

1. I am trying to get the Mateksys MATE-3901-LOX to work on UART 6 on the Kakute H7.
2. TX and RX go to UART RX and TX respectively, and power is on, lights are all.
3. I enabled MSP for UART 6 but obviously cannot select oflow / rangefinder under sensors if MSP is enabled
4. Doesn't seem to be working, sensor icons are not on.

Any idea how to enable developer debug mode so I can trace the logs?

**Reply by utrinque96** [reply]
Check this: https://github.com/iNavFlight/inav/issues/10566
